### PR TITLE
Add Dockerfiles and Vagrantfiles for dev/test images and environment

### PR DESCRIPTION
**Pull request by kshlm**

### CODE REVIEW

**Comment by kshlm** [comment]
@prashanthpai Can you review and merge this?

File: extras/dev-docker/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Docker image for GD2 development
+
+This docker image contains everything required to build and run GD2 during development.
+The image is available as a trusted build from Docker hub `gluster/glusterd2-dev`.
```
Comment (kshlm) [inline extras/dev-docker/README.md]: The images don't exist yet, as I need this to be merged before I set up the builds on Docker hub

File: extras/test-env/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# GD2 test environment - Docker + Vagrant
+
+This directory contains a `Dockerfile` to build a docker image with GD2 installed.
+A 'Vagrantfile' is provided which makes use of this docker image to setup a test env.
+
+## Docker image
+
+A trusted build 'gluster/glusterd2-test' is available from the Docker hub.
```
Comment (kshlm) [inline extras/test-env/README.md]: Same here.

File: extras/dev-docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM kshlm/vagrant-centos:latest
+
+MAINTAINER Kaushal M <kshlmster@gmail.com>
+
+ADD ./setup.sh /setup.sh
+RUN chmod +x /setup.sh
+RUN /setup.sh
+RUN rm /setup.sh
+
```
Comment (prashanthpai) [inline extras/dev-docker/Dockerfile]: Ok. No CMD line here. This makes sense though, as you use it in vagrant + docker + ssh.

File: extras/test-env/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+Vagrant.configure("2") do |config|
+  (1..4).each do |i|
+    config.vm.define "gd2test-#{i}" do |arch|
+      arch.vm.hostname = "gd2test-#{i}"
+      arch.vm.provider "docker" do |d|
+        d.image = 'gluster/glusterd2-test:latest'
+        d.name = "gd2test-#{i}"
+        d.has_ssh = true
+        d.remains_running = true
+        d.create_args = ['--privileged']
```
Comment (prashanthpai) [inline extras/test-env/Vagrantfile]: In later versions of this, I'd prefer we put some thought into not running this in privileged mode, considering glusterd2 is one binary without systemd. I'm wondering if actual deployment needs glusterd2 to do setxattr on bind mounted brick paths or if there's a better way around it. It should also play nice with selinux in unprivileged mode.

**Pull request merged**